### PR TITLE
Add support for passing JWT audience to defaultJWTClaimMapper

### PR DESCRIPTION
**Pull request by sergeybykov**
**What changed?**
Added support for passing JWT audience to `defaultJWTClaimMapper`.

**Why?**
`jwt-go` v4 started enforcing validation of JWT audience. For that, we need a way to pass audience argument to   `defaultJWTClaimMapper`, so that it can pass it to the JWT parser.

**How did you test it?**
Manually plus added a couple of unit tests.

**Potential risks**
I don't see risks here, as this is fixing the issue with `jwt-go` v4.

**Is hotfix candidate?**
Potentially

### CODE REVIEW

File: common/authorization/default_jwt_claim_mapper.go [hunk 1]
```diff
@@ -125,7 +125,18 @@ func (a *defaultJWTClaimMapper) extractPermissions(permissions []interface{}, cl
...
 
 func parseJWT(tokenString string, keyProvider TokenKeyProvider) (jwt.MapClaims, error) {
-	token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
+	return parseJWTWithAudience(tokenString, keyProvider, "")
+}
+
+func parseJWTWithAudience(tokenString string, keyProvider TokenKeyProvider, audience string) (jwt.MapClaims, error) {
+
+	var parser *jwt.Parser
+	if audience == "" {
```
Comment (feedmeapples) [inline common/authorization/default_jwt_claim_mapper.go:134]: should this also include `nil` case?

**Review by feedmeapples — APPROVED** [review]
lg, left one comment

File: common/authorization/default_jwt_claim_mapper.go (hunk 1 above)
Comment (mastermanu) [inline common/authorization/default_jwt_claim_mapper.go:134]: [Nit] Can we do strings.Trim(audience) == "" just to make sure empty space is also treated as no audience?

**Review by mastermanu — APPROVED** [review]

**Review by feedmeapples — APPROVED** [review]

File: common/authorization/interceptor.go [hunk 2]
```diff
@@ -45,6 +45,13 @@ type (
 	contextKeyAuthHeader   struct{}
 )
 
+type (
+	// AudienceGetter returns JWT audience for a given request
+	AudienceGetter interface {
```
Comment (underrun) [inline common/authorization/interceptor.go:50]: nit: the comment implies this is only for JWT and the name feels clunky - do we expect to have more JWT specific methods? maybe call this JwtMapper (or JWTMapper - depending on what our convention is)?

**Review by underrun — APPROVED** [review]

File: common/authorization/default_jwt_claim_mapper.go (hunk 1 above)
Comment (sergeybykov) [reply common/authorization/default_jwt_claim_mapper.go]: Strings can't be nil in Go.

Comment (sergeybykov) [reply common/authorization/default_jwt_claim_mapper.go]: Done.

File: common/authorization/interceptor.go (hunk 2 above)
Comment (sergeybykov) [reply common/authorization/interceptor.go]: Renamed to `JWTAudienceMapper`.

**Review by wxing1292 — APPROVED** [review]

**Comment by wxing1292** [comment]
@sergeybykov this PR should be part of v1.10.3?

**Review by alexshtin — APPROVED** [review]

**Comment by sergeybykov** [comment]
>@sergeybykov this PR should be part of v1.10.3?

I think so.

**Pull request merged**